modules/apps/transfer/keeper/keeper.go

```diff
@@ -169,8 +169,15 @@ func (k Keeper) SetTotalEscrowForDenom(ctx sdk.Context, coin sdk.Coin) {
 	}
 
 	store := ctx.KVStore(k.storeKey)
+	key := types.TotalEscrowForDenomKey(coin.Denom)
+
+	if coin.Amount.IsZero() {
+		store.Delete(key)
+		return
+	}
```

Comment:
needs code documentation on why we do this. I like having godoc and in-code comment